mlflow/pyspark/optuna/study.py

```diff
@@ -161,17 +181,70 @@ def __init__(
         mlflow.set_tracking_uri(self._mlflow_tracking_env)
         self.mlflow_client = MlflowClient()
 
-        self._study = optuna.create_study(
-            study_name=self.study_name, sampler=self.sampler, storage=self._storage
-        )
-        self._study_id = storage.get_study_id_from_name(self.study_name)
-        self._directions = self._storage.get_study_directions(self._study_id)
-
         if not isinstance(self._storage, MlflowStorage):
             raise ValueError(
                 f"MlflowSparkStudy only works with `MlflowStorage`. But get {type(self._storage)}."
             )
 
+        # Check if study exists and auto-resume if it does
+        if self._storage.study_exists(self.study_name):
+            # Load existing study
+            self._study = optuna.load_study(
+                study_name=self.study_name, sampler=self.sampler, storage=self._storage
+            )
+            self._study_id = self._storage.get_study_id_from_name(self.study_name)
+            self._is_resumed = True
+            _logger.info(
+                f"Resuming existing study '{self.study_name}' with {len(self._study.trials)} trials"
+            )
+        else:
+            # Create new study
+            self._study = optuna.create_study(
+                study_name=self.study_name, sampler=self.sampler, storage=self._storage
+            )
+            self._study_id = self._storage.get_study_id_from_name(self.study_name)
+            self._is_resumed = False
+            _logger.info(f"Created new study '{self.study_name}'")
+
+        self._directions = self._storage.get_study_directions(self._study_id)
+
+    @property
+    def is_resumed_study(self) -> bool:
+        """Check if this study was resumed from existing data.
+
+        Returns:
+            True if the study was resumed from existing data, False if it's a new study
+        """
+        return self._is_resumed
+
+    @property
+    def completed_trials_count(self) -> int:
+        """Number of completed trials in the study.
+
+        Returns:
+            Count of trials that have completed successfully
+        """
+        return len([t for t in self._study.trials if t.state == TrialState.COMPLETE])
+
+    def get_resume_info(self) -> ResumeInfo:
```

Comment:
```suggestion
    def get_resume_info(self) -> ResumeInfo | None
```

can we return None if it's not resumed?

Reply:
> It is mainly a helper function to check the status for loaded study.

I see, can directly check the study like:

```
assert study.is_removed == ...
assert study.best_params == ...
```